File: erlang/fizzbuzz.erl
-module(fizzbuzz).
-export([fizzbuzz_output/0]).

fizzbuzz_pattern(_, 0, 0) -> "fizzbuzz";
fizzbuzz_pattern(_, 0, _) -> "fizz";
fizzbuzz_pattern(_, _, 0) -> "buzz";
fizzbuzz_pattern(N, _, _) -> integer_to_list(N).

fizzbuzz(N) -> fizzbuzz_pattern(N, N rem 3, N rem 5).

fizzbuzz_output() -> lists:map(fun(N) -> io:format(fizzbuzz(N) ++ "\n") end, lists:seq(1,100)).

File: scala/fizzbuzz.scala
object fizzbuzz extends App {

  def fizzbuzzIndividual(number: Int): String = {
    number match{
      case n if n % 15 == 0 => "fizzbuzz"
      case n if n % 3 == 0 => "fizz"
      case n if n % 5 == 0 => "buzz"
      case n  => n.toString
    }
  }

  val fizzbuzz = (1 to 100).toList.map(fizzbuzzIndividual)
  println(fizzbuzz.mkString("\n"))
}
